### DISCUSSION TITLE
Use JAVA program to execute image comparison

### DISCUSSION THREAD

**Discussion by Mirai1337** [ImageMagick/ImageMagick#2117] [Help] [answered]
Hi,

So I wrote some code to execute the image comparison code but the system is returning error code 1 which I have no clue what it is. Can anyone help me to solve this one?

Below is the code

```java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Extract {

	public static void main(String[] args) {
		// TODO Auto-generated method stub
		Runtime rt = Runtime.getRuntime();
		try {
			Process pr = rt.exec("magick compare -metric PSNR C:\\Users\\Lenovo\\Desktop\\DO\\MonierStamp.png C:\\Users\\Lenovo\\Desktop\\DO\\MonierStamp2.png null");
			BufferedReader input = new BufferedReader(new InputStreamReader(pr.getInputStream()));
			 
            String line=null;

            while((line=input.readLine()) != null) {
                System.out.println(line);
            }
            int exitVal = pr.waitFor();
            System.out.println("Exited with error code "+exitVal);
		} catch (IOException | InterruptedException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
```

The output is:
Exited with error code 1

**Comment by dlemstra** [comment, marked as answer]
It is returning with a non zero exit code because the images are different. You could use that to decide what your next action will be.

**Reply by Mirai1337** [reply]
When I executed the command in CLI it returns some sort of number like 9.41245 something. May I know what that number is and why it's not appearing when I tried to print it out in the IDE?

**Comment by snibgo** [comment]
I don't know Java, but I suspect you are reading stdout from "magick compare", but you should read stderr.

**Reply by Mirai1337** [reply]
Hi, I did try it but still the same output
